src/otx/core/model/entity/base.py

```diff
@@ -410,6 +413,8 @@ def _callback(result: NamedTuple, idx: int) -> None:
                     self.model.await_any()
                 self.model.infer_async(im, user_data=idx)
             self.model.await_all()
+            print("FPS:", num_samples / (time.time() - start_time))
+            print("Time per sample:", (time.time() - start_time) / num_samples)
```

Comment:
Is this needed?

Reply:
In OTX 1.x, we log the FPS from the OV inferencer. This log is particularly valuable for Geti scenarios.

Reply:
If so, I recommend you to use https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.profilers.SimpleProfiler.html#lightning.pytorch.profilers.SimpleProfiler in this case as you know, otherwise, pass the FPS metric values to PyTorch Trainer's logger, not adding `print()` here. You might enhance https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algo/callbacks/iteration_timer.py to gather the FPS metric additionally.

Reply:
In addition, OTX 2 inference pipeline will not be used in Geti.

Reply:
SimpleProfiler is overly verbose in these situations, making it less useful. This is likely why we haven't made it the default option and instead opted to add a specific line of https://github.com/openvinotoolkit/training_extensions/blob/releases/2.0.0/src/otx/cli/__init__.py#L19

```py
def main() -> None:
    """Entry point for OTX CLI.

    This function is a single entry point for all OTX CLI related operations:
    """
    start = time()
    OTXCLI()
    dt = timedelta(seconds=time() - start)
    print(f"Elapsed time: {dt}")
```

The underlying concept is quite similar.

Plus, the impact of the iteration time on printing to stdout is minimal, especially considering the large batch size specified in openvino_model.yaml.

However, if you're not a fan of it, I can remove it.

Reply:
I'm cautioning against the additive manner in which new implementation is being added without consideration for existing ones. Functionally, batch iteration time and batch throughput are quite similar. Therefore, instead of increasing code entropy, utilize existing functionality or extend it if possible. The current code structure ensures that throughput is logged only for OVModel, excluding OTXModel. Why does OTXModel lack this throughput reporting, and why is this code inserted only in OVModel? If you were to leave the maintenance in the future or a new person unfamiliar with the history were to join, it would be a struggling bar to understand this.

Reply:
That's a fair point. I'm removing it and exploring a more holistic approach. The rationale behind not adding it to OTXModel is that Geti exclusively runs inference/predictions on OV models, making it a valuable metric. However, if Geti won't utilise OTX 2.x for inference or prediction then it's not serving much of a purpose.